db/db_impl/db_impl.h

```diff
@@ -1187,7 +1187,13 @@ class DBImpl : public DB {
   // WriteToWAL need different synchronization: log_empty_, alive_log_files_,
   // logs_, logfile_number_. Refer to the definition of each variable below for
   // more description.
-  mutable InstrumentedMutex mutex_;
+  //
+  // `mutex_` can be a hot lock in some workloads, so it deserves dedicated
+  // cachelines.
+  ALIGN_AS(CACHE_LINE_SIZE) mutable InstrumentedMutex mutex_;
```

Comment:
Or can we make the following `default_cf_handle_` also aligned on cache line size.
Both the approach in the PR and the above proposal require that the members are in the same access block, e.g. `private` or `public` or `protected`. I hope we can enforce this and not have to worry about (accidental and unlikely) following change:
```
ALIGN_AS(CACHE_LINE_SIZE) mutable InstrumentedMutex mutex_;
private:
 char padding[(CACHE_LINE_SIZE -
                (sizeof(InstrumentedMutex) % CACHE_LINE_SIZE)) %
               CACHE_LINE_SIZE] ROCKSDB_FIELD_UNUSED;
```
How about wrap the mutex_ and padding in another struct? This may incur more changes, so I would not strongly advocate it.

Reply:
Let me check if it's possible to make an AlignedInstrumentedMutex structure without too much changes.

Reply:
Done. C++17 made this much easier. Thanks for the review!